src/rulesets/LinearAlgebra/structured.jl

```diff
@@ -32,6 +32,9 @@ end
 #####
 
 function rrule(::Type{<:Diagonal}, d::AbstractVector)
+    function Diagonal_pullback(x::AbstractThunk)
+        return Diagonal_pullback(unthunk(x))
+    end
```

Comment:
```suggestion
```

Not needed see https://github.com/JuliaDiff/ChainRules.jl/pull/282#issuecomment-708452476